### PR TITLE
[dagster-dbt] add `with_snowflake/bigquery_insights` chainable method

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Adds a `with_snowflake_insights` and `with_bigquery_insights` methods that can be chained to a `.stream()` call, simplifying using insights alongside other deferred fetches:

Before:

```python
@dbt_assets(manifest=snowflake_manifest_path)
def jaffle_shop_dbt_assets(
    context: AssetExecutionContext,
    dbt: DbtCliResource,
):
    dbt_cli_invocation = dbt.cli(["build", "--profile", "unittest"], context=context)
    yield from dbt_with_snowflake_insights(context, dbt_cli_invocation)

```

Now:

```python
@dbt_assets(manifest=snowflake_manifest_path)
def jaffle_shop_dbt_assets(
    context: AssetExecutionContext,
    dbt: DbtCliResource,
):
    yield from dbt.cli(
        ["build", "--profile", "unittest"], context=context
    ).stream().with_snowflake_insights()
```

## Test Plan

https://github.com/dagster-io/internal/pull/10731

### CODE REVIEW

**Comment by benpankow** [comment]
* **#23183** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23183?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by rexledesma — DISMISSED** [review]
Can we just have one function, `with_insights`? We can determine whether the target is `snowflake` or `bigquery` from the execution manifest

Also, instead of an import time error, maybe we could raise a warning instead, so core execution doesn't fail? We should also add a test in the OSS context (the warning is thrown since `dagster-cloud` isn't installed)

**Comment by benpankow** [comment]
> Can we just have one function, with_insights? We can determine whether the target is snowflake or bigquery from the execution manifest

Good call; updated.

> Also, instead of an import time error, maybe we could raise a warning instead, so core execution doesn't fail? We should also add a test in the OSS context (the warning is thrown since dagster-cloud isn't installed)

I'd prefer that the code explicitly errors here just so there's no silent failure or other unexpected behavior - in the case that a user is conditionally using different warehouses, I think they can just conditionally call this method.

File: python_modules/libraries/dagster-dbt/dagster_dbt/core/resource.py [hunk 1]
```diff
@@ -802,6 +802,83 @@ def _threadpool_wrap_map_fn() -> (
...
+                    record_observation_usage=record_observation_usage,
+                ),
+                dbt_cli_invocation=self._dbt_cli_invocation,
+            )
+        elif adapter_type == "bigquery":
+            try:
+                from dagster_cloud.dagster_insights import dbt_with_bigquery_insights
+            except ImportError as e:
+                raise DagsterInvalidPropertyError(
+                    "The `dagster_cloud` library is required to use the `with_bigquery_insights`"
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt/core/resource.py:863]:
```suggestion
                    "The `dagster_cloud` library is required to use the `with_insights`"
```

**Review by rexledesma — APPROVED** [review]
Seems like we should just universally have the config check enabled? That way, we can remove it as an argument. But we should probably ensure that check is done at the start of the dbt cli invocation so that things fail fast.

Not a blocker to get this initial scaffold in though

**Pull request merged**